src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,290 @@
+            if (t != null) {
+                System.out.println(++listIndex + ". " + t);
+            }
+        }
+    }
+
+
+    private static void writeTasksToFile(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task t : tasks) {
+            if (t != null) {
+                fw.write(t.saveFormat() + System.lineSeparator());
+            }
+        }
+
+        fw.close();
+    }
+
+    private static void save(String filename) throws IOException {
+        try {
+            writeTasksToFile(filename);
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void load(String filename) {
+        File f = new File(filename);
+
+        if (f.exists()) {
+            try {
+                loadFileContents(f);
+            } catch (FileNotFoundException e) {
+                System.out.println(FILE_NOT_FOUND_MESSAGE);
+            }
+        }
+    }
+
+    private static void loadFileContents(File f) throws FileNotFoundException {
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            processFileContents(s.nextLine());
+        }
+    }
+
+    private static void processFileContents(String line){
+        String[] words = line.split("\\|");
+        String type = words[0];
+        boolean isDone = false;
+        if (Integer.valueOf(words[1]) == 1) {
+            isDone = true;
+        }
+        switch (type) {
+        case "T":
+            loadTodo(words, isDone);
+            break;
+        case "E":
+            loadEvent(words, isDone);
+            break;
+        case "D":
+            loadDeadline(words, isDone);
+            break;
+        default:
+            // unknown char error
+        }
+    }
+
+    private static void loadDeadline(String[] words, boolean isDone) {
+        Deadline deadline = new Deadline(words[2], words[3]);
+        if(isDone){
+            deadline.markAsDone();
+        }
+        tasks.add(deadline);
+    }
+
+    private static void loadEvent(String[] words, boolean isDone) {
+        Event event = new Event(words[2], words[3], words[4]);
+        if(isDone){
+            event.markAsDone();
+        }
+        tasks.add(event);
+    }
+
+    private static void loadTodo(String[] words, boolean isDone) {
+        Todo todo = new Todo(words[2]);
+        if(isDone){
+            todo.markAsDone();
+        }
+        tasks.add(todo);
+    }
+
+    private static Deadline getDeadline(String[] words) {
+        String line;
+        line = words[1]; // to remove the command
+        String[] deadlineDetails = line.split(BY_DELIMITER);
+        Deadline d = new Deadline(deadlineDetails[0], deadlineDetails[1]);
+        return d;
+    }
+
+    private static Event getEvent(String[] words) {
+        String line;
+        line = words[1]; // to remove the command
```

Comment:
Consider having more explainable names - what is the difference between `line` and `words`?

https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain